pkg/services/rocketchat.go

```diff
@@ -0,0 +1,134 @@
+package services
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
+	"github.com/RocketChat/Rocket.Chat.Go.SDK/rest"
+	"net/url"
+	"regexp"
+	"strings"
+	texttemplate "text/template"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type RocketChatNotification struct {
+	Attachments string `json:"attachments,omitempty"`
+}
+
+func (n *RocketChatNotification) GetTemplater(name string, f texttemplate.FuncMap) (Templater, error) {
+	rocketChatAttachments, err := texttemplate.New(name).Funcs(f).Parse(n.Attachments)
+	if err != nil {
+		return nil, err
+	}
+	return func(notification *Notification, vars map[string]interface{}) error {
+		if notification.RocketChat == nil {
+			notification.RocketChat = &RocketChatNotification{}
+		}
+		var rocketChatAttachmentsData bytes.Buffer
+		if err := rocketChatAttachments.Execute(&rocketChatAttachmentsData, vars); err != nil {
+			return err
+		}
+
+		notification.RocketChat.Attachments = rocketChatAttachmentsData.String()
+
+		return nil
+	}, nil
+}
+
+type RocketChatOptions struct {
+	Alias              string   `json:"alias"`
+	Email              string   `json:"email"`
+	Password           string   `json:"password"`
+	Icon               string   `json:"icon"`
+	Avatar             string   `json:"avatar"`
+	Token              string   `json:"token"`
+	ServerUrl          string   `json:"serverUrl"`
+	SigningSecret      string   `json:"signingSecret"`
+	Channels           []string `json:"channels"`
+	InsecureSkipVerify bool     `json:"insecureSkipVerify"`
+}
+
+type rocketChatService struct {
+	opts RocketChatOptions
+}
+
+var validEmoji = regexp.MustCompile(`^:.+:$`)
+
+func NewRocketChatService(opts RocketChatOptions) NotificationService {
+	return &rocketChatService{opts: opts}
+}
+
+func (r *rocketChatService) Send(notification Notification, dest Destination) error {
+	serverUrl, err := url.Parse(r.opts.ServerUrl)
+	if err != nil {
+		return err
+	}
+
+	rl := rest.NewClient(serverUrl, false)
+
+	credentials := models.UserCredentials{Email: r.opts.Email, Password: r.opts.Password}
+	err = rl.Login(&credentials)
+	if err != nil {
+		return err
+	}
+
+	message := models.PostMessage{Alias: r.opts.Alias, Text: notification.Message}
+	// It's a channel
+	if strings.ContainsAny(dest.Recipient, "@#") {
```

Comment:
I guess you can use `strings.HasPrefix` instead of `strings.ContainsAny`.

Reply:
Thanks. It's a cleaner solution that I have. Fixed